packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -256,140 +258,268 @@ export function isAssignable(
 	return ans
 }
 
-interface CheckerTreeDefinitionNode<T> {
-	typeDef: SimplifiedMcdocTypeNoUnion
-	runtimeNode: CheckerTreeRuntimeNode<T>
-	children: CheckerTreeEntryNode<T>[]
-	errors: McdocCheckerError<T>[]
+interface CheckerTreeNode<T> {
+	parent: CheckerTreeRuntimeNode<T> | undefined
+
+	runtimeKey: RuntimeNode<T> | undefined
+	possibleValues: CheckerTreeRuntimeNode<T>[]
+	validDefinitions: CheckerTreeDefinitionNode<T>[]
+	condensedErrors: CheckerTreeError<T>[][]
 }
+
+interface CheckerTreeError<T> {
+	error: McdocCheckerError<T>
+	definitionNode: CheckerTreeDefinitionNode<T> | undefined
+}
+
 interface CheckerTreeRuntimeNode<T> {
+	entryNode: CheckerTreeNode<T>
+	children: CheckerTreeNode<T>[]
 	node: RuntimeNode<T>
-	entryNode: CheckerTreeEntryNode<T>
-	validDefinitions: CheckerTreeDefinitionNode<T>[]
-	/// Each outer entry represents a layer of siblings
-	/// Each inner entry represents a specific condensed error within that layer.
-	condensedChildErrors?: McdocCheckerError<T>[][]
 }
-interface CheckerTreeEntryNode<T> {
+
+interface CheckerTreeDefinitionNode<T> {
 	parent: CheckerTreeDefinitionNode<T> | undefined
-	runtimeKey?: RuntimeNode<T>
-	possibleRuntimeValues: CheckerTreeRuntimeNode<T>[]
-	typeDef: McdocType | undefined
+	entryNode: CheckerTreeNode<T>
```

Comment:
This field seems to be unused

Reply:
I think I like keeping a reference back to the effective owner of this node, even though it is actually indeed unused in this case.

Reply:
(this changed to `runtimeNode` now with a similar semantic, due to validDefinitions being moved to CheckerTreeRuntimeNode. Probably still unused though, so it could be removed if you prefer that still)